src/cookbook/gradual-compositionapi-migration.md

```diff
@@ -0,0 +1,258 @@
+
+Due to the flexibility of the Composition API, you may need to be more explicit about organization so that future maintainers can quickly understand intended purpose of pieces of the application.
+
+## Refactor Your Mixins
+
+If you've made use of mixins for reusable logic across components, it's recommended that you start your refactor here. Mixins translate very directly.
+
+### Base Example
+
+Let's say we had this very small toggle mixin in `mixins/toggle.js`:
+
+```js
+const toggle = {
+  data() {
+    return {
+      isShowing: false
+    }
+  },
+  methods: {
+    toggleShow() {
+      this.isShowing = !this.isShowing
+    }
+  }
+}
+```
+
+And it was being used in a modal component in `components/modal.vue`:
+
+```html
+<template>
+  <div>
+    <h3>Let's trigger this here modal!</h3>
+
+    <button @click="toggleShow">
+      <span v-if="isShowing">Hide child</span>
+      <span v-else>Show child</span>
+    </button>
+
+    <div v-if="isShowing" class="modal">
+      <h2>Here I am!</h2>
+      <button @click="toggleShow">Close</button>
+    </div>
+  </div>
+</template>
+```
+
+```js
+<script>
+  import { toggle } from '@/mixins/toggle';
+
+  export default {
+    mixins: [toggle]
+  }
+</script>
+```
+
+We could refactor this by creating a `composables` folder and create `composables/useToggle.js` instead. We suggest using a directory named `composables` so that you can communicate that this is being used slightly differently from a component, it's reusable logic that you can consume.
+
+```js
+export function useToggle() {
+  const state = reactive({
+    isShowing: false
+  })
+
+  function toggleShow() {
+    state.isShowing = !state.isShowing
+  }
+
+  return {
+    ...toRefs(state),
+    toggleShow
+  }
+}
+```
+
+And we can refactor our component as follows:
+
+```html
+<template>
+  <div>
+    <h3>Let's trigger this here modal!</h3>
+
+    <button @click="toggleShow">
+      <span v-if="isShowing">Hide child</span>
+      <span v-else>Show child</span>
+    </button>
+
+    <div v-if="isShowing" class="modal">
+      <h2>Here I am!</h2>
+      <button @click="toggleShow">Close</button>
+    </div>
+  </div>
+</template>
+```
+
+```js
+<script>
+  import { useToggle } from "@/composables/useToggle.js";
+
+  export default {
+    setup() {
+      const { isShowing, toggleShow } = useToggle();
+
+      return {
+        isShowing,
+        toggleShow
+      };
+    },
+  }
+</script>
+```
+
+<iframe src="https://codesandbox.io/embed/refactor-mixin-to-composition-api-hnloh?fontsize=14&hidenavigation=1&theme=dark"
```

Comment:
Ah, yeah, I changed the sandbox to be more clear what the focus was. I'll update that. Thanks!

And yeah, it's using Vue 2 intentionally because we're showing how to gradually update from an existing Vue 2 application, not from Vue 3 greenfield.